src/main/java/cloud/fogbow/ras/core/OrderController.java

```diff
@@ -145,9 +142,8 @@ public void deleteOrder(Order order) throws FogbowException {
                     remoteCloudConnector.deleteInstance(order);
                     // This is just to make sure the remote provider order will be moved to the remoteProviderOrders
                     // list (if it is not already there), since PENDING orders belong to this list.
-                    OrderStateTransitioner.transition(order, OrderState.PENDING);
-                    // This changes the state of the order without removing it from the remoteProviderOrders list
-                    order.setOrderState(OrderState.ASSIGNED_FOR_DELETION);
+                    // TODO(chico) - Check with team.
+                    OrderStateTransitioner.transitionToRemoteList(order, OrderState.ASSIGNED_FOR_DELETION);
```

Comment:
I see your point, but it is still a bit smelly. Let us leave as you suggest, for now, since I cannot think of a better solution right now. Please, remove TODO comment.

Reply:
Ok. done!